### ISSUE TITLE
[Feature Request] Add an advanced setting to specify the executable for youtube-dl

### ISSUE THREAD

**Issue by ehoogeveen-medweb** [clsid2/mpc-hc#1175]
I would like to specify an alternative executable for loading links with youtube-dl. Specifically I use the fork [yt-dlp](https://github.com/yt-dlp/yt-dlp) which adds some additional features and workarounds. Simply being able to s/youtube-dl/yt-dlp/ would be enough; currently the string `youtube-dl` appears to be hardcoded in the invocations in `src/mpc-hc/MainFrm.cpp` and `src/mpc-hc/YoutubeDL.cpp`.

**Comment by lifegpc** [comment]
You can just rename that executable to youtube-dl.exe and put it in the directory where mpc-hc.exe or mpc-hc64.exe located.

**Comment by ehoogeveen-medweb** [comment]
Well, I guess that's true. But I use pip to keep it up-to-date so I would rather use something built into mpc-hc.

**Comment by ale5000-git** [comment]
You can create a symbolic link to avoid rename and move.

**Comment by ale5000-git** [comment]
You can also make it update itself with `youtube-dl.exe -U`

**Comment by ehoogeveen-medweb** [comment]
Oh yeah, I guess Windows does support symlinks these days. The following worked for me, though I'd still prefer to have it as an option in mpc-hc:
`mklink "%LocalAppData%\Programs\Python\Python39\Scripts\youtube-dl.exe" "%LocalAppData%\Programs\Python\Python39\Scripts\yt-dlp.exe"`

**Comment by clsid2** [comment]
Option has been added.

**Comment by ehoogeveen-medweb** [comment]
Thank you!

**Comment by ghost** [comment]
EDIT: Using the Options… > Advanced > YDLExePath UI and setting the value "C:\Python39\Scripts\yt-dlp.exe" will only work during the instance of MPC-HC. Closing then reopening MPC-HC will clear the setting.
Actually, MPC-HC doesn't even write to the registry key HKEY_CURRENT_USER\SOFTWARE\MPC-HC\MPC-HC\Settings\YDLExePath. The setting is only valid in memory. Setting the registry key directly with type string (e.g. "YDLExePath"="C:\Python39\Scripts\yt-dlp.exe") will make it work during program launch and runtime.
TL;DR changing setting in UI doesn't work.
@ehoogeveen-medweb please confirm this bug behaviour.

This doesn't work with environment variables, only hard paths.
C:\Python39\Scripts\yt-dlp.exe works.
%HOMEDRIVE%\Python39\Scripts\yt-dlp.exe doesn't.
@clsid2 This is not really important of an issue, but might as well mention it.

**Comment by clsid2** [comment]
I have fixed the saving of the setting.

**Comment by clsid2** [comment]
Environment variables are now also expanded.
